### ISSUE TITLE
Segmentation fault when updating argocd_project resources

### ISSUE THREAD

**Issue by albgus** [oboukili/terraform-provider-argocd#130]
Hello,

I am seeing this error when attempting to update a bunch of imported argocd_project resources using this provider

```
Stack trace from the terraform-provider-argocd_v2.2.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x21249c3]

goroutine 118 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDProjectUpdate(0x2b105d8, 0xc000c2d560, 0xc000ada100, 0x24d0060, 0xc0005c70e0, 0xc0007ac070, 0x6fb0ea, 0xc000d74b20)
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_project.go:252 +0xac3
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000779ce0, 0x2b10568, 0xc000ec3e40, 0xc000ada100, 0x24d0060, 0xc0005c70e0, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:364 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000779ce0, 0x2b10568, 0xc000ec3e40, 0xc0003ffe30, 0xc000d74b20, 0x24d0060, 0xc0005c70e0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0008b82e8, 0x2b10568, 0xc000ec3e40, 0xc000e9f400, 0xc000ec3e40, 0x25f46e0, 0xc0010ad900)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000d75540, 0x2b10610, 0xc000ec3e40, 0xc0003ffb90, 0xc000d75540, 0xc000b07ba0, 0x46cfa7)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x25f46e0, 0xc000d75540, 0x2b10610, 0xc0010ad9b0, 0xc000ea8ea0, 0x0, 0x2b10610, 0xc0010ad9b0, 0xc0010b2000, 0xc93)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004a9340, 0x2b29938, 0xc00060a480, 0xc001084240, 0xc000814960, 0x3e255e0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.38.0/server.go:1286 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0004a9340, 0x2b29938, 0xc00060a480, 0xc001084240, 0x0)
	google.golang.org/grpc@v1.38.0/server.go:1609 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000554bc0, 0xc0004a9340, 0x2b29938, 0xc00060a480, 0xc001084240)
	google.golang.org/grpc@v1.38.0/server.go:934 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd

Error: The terraform-provider-argocd_v2.2.0 plugin crashed!
```

The terraform resource used to create this looks like this:
```

resource "argocd_project" "this" {
  metadata {
    name      = local.argocd_project_name
    namespace = "argocd"
    labels    = {
      environment = var.target.environment
    }
  }

  spec {
    source_repos = var.source_repos

    role {
      name     = "full"
      policies = [
        "p, proj:${local.argocd_project_name}:full, applications, get, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:full, applications, update, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:full, applications, override, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:full, applications, sync, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:full, applications, action/*, ${local.argocd_project_name}/*, allow",
      ]
      groups = var.full_access_group_ids
    }

    role {
      name     = "sync"
      policies = [
        "p, proj:${local.argocd_project_name}:sync, applications, get, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:sync, applications, sync, ${local.argocd_project_name}/*, allow",
        "p, proj:${local.argocd_project_name}:sync, applications, action/restart, ${local.argocd_project_name}/*, allow",
      ]
      groups = var.sync_group_ids
    }

    role {
      name     = "read"
      policies = [
        "p, proj:${local.argocd_project_name}:read, applications, get, ${local.argocd_project_name}/*, allow",
      ]
      groups = var.read_group_ids
    }

    dynamic "destination" {
      for_each = toset(local.namespaces)
      content {
        server    = var.target.argocd_destination_server
        namespace = destination.value
      }
    }

    dynamic "cluster_resource_whitelist" {
      for_each = var.argocd_cluster_resource_whitelist

      content {
        group = cluster_resource_whitelist.value.group
        kind  = cluster_resource_whitelist.value.kind
      }
    }

    orphaned_resources {
      warn = false
      ignore {
        group = "rbac.authorization.k8s.io"
        kind  = "RoleBinding"
      }
    }
  }

  # FIXME: A bug is causing the provider to always configure a orphaned_resources block with empty configuration, which enables orphaned resource tracking. Therefore attempting to set it up with defaults that
  # ignores meta resources created by Rancher, but also ignores changes in case we find a need to disable it in ArgoCD.
  lifecycle {
    ignore_changes = [
      spec[0].orphaned_resources
    ]
  }
}
```

Given the line mentioned in the stack trace, I presume it has something to do with JWT tokens, but I haven't been able to figure out a workaround yet.

https://github.com/oboukili/terraform-provider-argocd/blob/master/argocd/resource_argocd_project.go#L252

**Comment by vj396** [comment]
I also am running into the same issue,

but, for me, I had imported an already created project (using UI) into terraform.

This issue does not present itself when I have created a project using this provider and updated the project using terraform only.

**Comment by MrLuje** [comment]
Fixed in v2.2.7
